mlir/include/mlir/Dialect/Bufferization/Transforms/Passes.h

```diff
@@ -134,8 +135,9 @@ func::FuncOp buildDeallocationLibraryFunction(OpBuilder &builder, Location loc,
 LogicalResult deallocateBuffers(Operation *op);
 
 /// Run ownership basedbuffer deallocation.
-LogicalResult deallocateBuffersOwnershipBased(FunctionOpInterface op,
-                                              bool privateFuncDynamicOwnership);
+LogicalResult deallocateBuffersOwnershipBased(
+    FunctionOpInterface op,
+    const DeallocationOptions &options = DeallocationOptions());
```

Comment:
Maybe double check, but I think default values should not be bound to const references.

Reply:
If I understand the standard correctly, this should be fine. The livetime of the temporary object will be extended to that of the const reference bound to. (it would be illegal for non-const references though)